### PR TITLE
Fix to readGlobalFile (resolves #468)

### LINKED ISSUE #468: job.fileStore.readGlobalFile() should have an option to write to a specific temp dir obtained using job.fileStore.getLocalTempDir()
if a tool requires n inputs, we need them all to be in the same dir so the dir can be mounted to /data in a docker image. We cannot assume that job.fileStore.readGlobalFile() as it is right now will read all n files into the same directory

### PR DESCRIPTION
**Pull request by benedictpaten**
This fix follows the discussion with Arjun and Hannes.

Resolves #468

### CODE REVIEW

File: src/toil/job.py [hunk 1]
```diff
@@ -478,22 +478,49 @@ def writeGlobalFileStream(self, cleanup=False):
...
+        def readGlobalFile(self, fileStoreID, userPath=None):
             """
-            Returns a path to a local copy of the file keyed by fileStoreID. 
-            The file will be read only. 
+            Returns an absolute path to a local, temporary copy of the file 
+            keyed by fileStoreID. 
+            
+            *The returned file will be read only (have permissions 444).* 
+            
+            userPath is a path to the name of file to which the global file will be
```
Comment (hannes-ucsc) [inline src/toil/job.py]: Love the extensive documentation but we should use `:param:` tags if possible

**Comment by hannes-ucsc** [comment]
With this change we should also roll back my workaround in the CWL runner.

**Comment by hannes-ucsc** [comment]
And by my workaround I mean #466.

**Comment by benedictpaten** [comment]
Closing in favor of PR #477, which use a properly named branch.

**Pull request closed without merging**